tests/lib/rules/jsx-sort-props.js

```diff
@@ -192,12 +192,12 @@ ruleTester.run('jsx-sort-props', rule, {
     },
     {
       code: '<App {...this.props} c="a" a="c" b="b" />;',
-      output: '<App a="c" b="b" c="a" {...this.props} />;',
+      output: '<App {...this.props} a="c" b="b" c="a" />;',
       errors: 2
     },
     {
       code: '<App d="d" b="b" {...this.props} c="a" a="c" />;',
-      output: '<App a="c" b="b" c="a" d="d" {...this.props} />;',
+      output: '<App a="c" b="b" {...this.props} c="a" d="d" />;',
```

Comment:
sorry if i wasn't clear - i'm afraid this is also unsafe. Any element with a spread prop *can not be touched in any way* that would alter overrides of the spread object. So, in `<App a="c" b="b" {...this.props} c="a" d="d" />`, you could sort the list of `[a, b]`, and the list of `[c, d]` - but a and b must always remain *before* `...this.props`, and `c` and `c` must remain after it.